packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -4314,3 +4354,252 @@ class InputDecorationTheme with Diagnosticable {
+      : super();
+
+  final BuildContext context;
+
+  @override
+  TextStyle? get inputStyle => Theme.of(context).textTheme.subtitle1;
+
+  @override
+  Color? get fillColor {
+    // dark theme: 10% white
+    // light theme: 4% black
+    switch (Theme.of(context).brightness) {
+      case Brightness.dark: return const Color(0x1AFFFFFF);
+      case Brightness.light:return const Color(0x0A000000) ;
+    }
+  }
+
+  @override
+  Color? get disabledFillColor {
+    // dark theme: 5% white
+    // light theme: 2% black
+    switch (Theme.of(context).brightness) {
+      case Brightness.dark:
+        return const Color(0x0DFFFFFF);
+      case Brightness.light:
+        return const Color(0x05000000) ;
+    }
+  }
+
+  @override
+  Color? get iconColor => MaterialStateColor.resolveWith((Set<MaterialState> states) {
+    if (states.contains(MaterialState.disabled) && !states.contains(MaterialState.focused)) {
+      return Theme.of(context).disabledColor;
+    }
+    if (states.contains(MaterialState.focused)) {
+      return Theme.of(context).colorScheme.primary;
+    }
+    switch (Theme.of(context).brightness) {
+      case Brightness.dark:
+        return Colors.white70;
+      case Brightness.light:
+        return Colors.black45;
+    }
+  });
+
+  @override
+  Color? get prefixIconColor => MaterialStateColor.resolveWith((Set<MaterialState> states) {
+    if (states.contains(MaterialState.disabled) && !states.contains(MaterialState.focused)) {
+      return Theme.of(context).disabledColor;
+    }
+    if (states.contains(MaterialState.focused)) {
+      return Theme.of(context).colorScheme.primary;
+    }
+    switch (Theme.of(context).brightness) {
+      case Brightness.dark:
+        return Colors.white70;
+      case Brightness.light:
+        return Colors.black45;
+    }
+  });
+
+  @override
+  Color? get suffixIconColor => MaterialStateColor.resolveWith((Set<MaterialState> states) {
+    if (states.contains(MaterialState.disabled) && !states.contains(MaterialState.focused)) {
+      return Theme.of(context).disabledColor;
+    }
+    if (states.contains(MaterialState.focused)) {
+      return Theme.of(context).colorScheme.primary;
+    }
+    switch (Theme.of(context).brightness) {
+      case Brightness.dark:
+        return Colors.white70;
+      case Brightness.light:
+        return Colors.black45;
+    }
+  });
+
+}
+
+// BEGIN GENERATED TOKEN PROPERTIES - InputDecorator
+
+// Do not edit by hand. The code between the "BEGIN GENERATED" and
+// "END GENERATED" comments are generated from data in the Material
+// Design token database by the script:
+//   dev/tools/gen_defaults/bin/gen_defaults.dart.
+
+// Token database version: v0_101
+
+// Generated version v0_101
+class _InputDecoratorDefaultsM3 extends InputDecorationTheme {
+  const _InputDecoratorDefaultsM3(this.context)
+    : super();
+
+  final BuildContext context;
+
+  @override
+  Color? get fillColor => Theme.of(context).colorScheme.surfaceVariant;
```

Comment:
Why not use a MaterialStateColor for fillColor and disabledFillColor?

Reply:
Done